pkg/api/types.go

```diff
@@ -2700,6 +2725,10 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity
 
+	// ServiceAffinityConfig contains the configurations of session affinity.
+	// +optional
+	ServiceAffinityConfig *ServiceAffinityConfig
```

Comment:
I agree we should consider it more.

In my previous implementaion, it was

```
sessionAffinity: ClientIP
clientIPAffinityConfig:
  timeoutSeconds: 60
```

And from your previous comment,

> I think you want a struct, for future safety.

>type ServiceAffinityConfig struct {
  // only one of the following may be set
  ClientIPConfig *ServiceAffinityConfigClientIP
}

>type ServiceAffinityConfigClientIP struct {
  TimeoutSeconds int
}
Then embed ServiceAffinityConfig

I guess you think `clientIPAffinityConfig` was not future-safe enough. So, I introduced a 2-level nesting.

```
sessionAffinity: ClientIP
serviceAffinityConfig:
  clientIPConfig:
    timeoutSeconds: 60
```

I think the difference is `serviceAffinityConfig` vs. `clientIPAffinityConfig`. I  think we should consider the API expandability. For example, if we introduce a new sessionAffinity type, such as "UserName". Perhaps we will to create a new type `userNameAffinityConfig` which has a `timeoutSeconds` field.

2 levels of nesting would be 

```
sessionAffinity: UserName
serviceAffinityConfig:
  userNameAffinityConfig:
    timeoutSeconds: 60
```

embedded would be:

```
sessionAffinity: UserName
userNameAffinityConfig:
  timeoutSeconds: 60
```



It seems **embedded** way looks more clean and it seems `serviceAffinityConfig` is unnecessary.